pkg/kfapp/kustomize/kustomize.go

```
@@ -350,36 +346,84 @@ func (kustomize *kustomize) Delete(resources kftypesv3.ResourceEnum) error {
+					kustomize.kfDef.ClusterName, ctx.Cluster)
 			}
 		}
-		if msg != "" {
-			if forceDelete {
-				log.Warnf(msg)
-			} else {
+	}
+	if msg != "" {
+		if forceDelete {
+			log.Warnf(msg)
+		} else {
+			return &kfapisv3.KfError{
+				Code:    int(kfapisv3.INVALID_ARGUMENT),
+				Message: msg,
+			}
+		}
+	}
+	kustomize.initK8sClients()
+	kubeclient, err := client.New(kustomize.restConfig, client.Options{})
+	if err != nil {
+		return &kfapisv3.KfError{
+			Code:    int(kfapisv3.INTERNAL_ERROR),
+			Message: fmt.Sprintf("error initializing k8s client Error %v", err),
+		}
+	}
+
+	// Delete in reverse application order
+	kustomizeDir := path.Join(kustomize.kfDef.Spec.AppDir, outputDir)
+	for idx := range kustomize.kfDef.Spec.Applications {
+		app := &kustomize.kfDef.Spec.Applications[len(kustomize.kfDef.Spec.Applications)-1-idx]
+		log.Infof("Deleting application %v", app.Name)
+		resMap, err := EvaluateKustomizeManifest(path.Join(kustomizeDir, app.Name))
+		if err != nil {
+			log.Errorf("error evaluating kustomization manifest for %v Error %v", app.Name, err)
+			return &kfapisv3.KfError{
+				Code:    int(kfapisv3.INTERNAL_ERROR),
+				Message: fmt.Sprintf("error evaluating kustomization manifest for %v Error %v", app.Name, err),
+			}
+		}
+		yamlBytes, err := resMap.AsYaml()
+		if err != nil {
+			return &kfapisv3.KfError{
+				Code:    int(kfapisv3.INTERNAL_ERROR),
+				Message: fmt.Sprintf("error evaluating kustomization manifest for %v Error %v", app.Name, err),
+			}
+		}
+		resources, err := utils.SplitYAML(yamlBytes)
+		if err != nil {
+			return &kfapisv3.KfError{
+				Code:    int(kfapisv3.INTERNAL_ERROR),
+				Message: fmt.Sprintf("error splitting yaml: %v", err),
+			}
+		}
+		for _, r := range resources {
+			err := utils.DeleteResource(r, kubeclient, 5*time.Minute)
+			if err != nil {
 				return &kfapisv3.KfError{
-					Code:    int(kfapisv3.INVALID_ARGUMENT),
-					Message: msg,
+					Code:    int(kfapisv3.INTERNAL_ERROR),
```

Should we continue to delete the rest as a best effort or quit immediately?

---

That's a good question :thinking: 
I usually think it's good to fail early.
On the other hand, kubectl tries to delete everything regardless.
We may want to stay close to kubectl behavior as in the future people will probably use it.